cmd/prom-migrator/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/timescale/promscale/pkg/log"
+	plan "github.com/timescale/promscale/pkg/migration-tool/planner"
+	"github.com/timescale/promscale/pkg/migration-tool/reader"
+	"github.com/timescale/promscale/pkg/migration-tool/writer"
+)
+
+type config struct {
+	name               string
+	mint               int64
+	maxt               int64
+	readURL            string
+	writeURL           string
+	writerReadURL      string
+	progressEnabled    bool
+	progressMetricName string
+}
+
+func main() {
+	conf := new(config)
+	parseFlags(conf, os.Args[1:])
+	if err := log.Init(log.Config{Format: "logfmt", Level: "debug"}); err != nil {
+		_, _ = fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	log.Info("msg", fmt.Sprintf("%v+", conf))
+	if err := validateConf(conf); err != nil {
+		log.Error("msg", "could not parse flags", "error", err)
+		os.Exit(1)
+	}
+
+	planner := &plan.Plan{
+		Mint:    conf.mint,
+		Maxt:    conf.maxt,
+		JobName: conf.name,
+		// Progress metric configs.
+		ProgressMetricName:        conf.progressMetricName,
+		ProgressEnabled:           conf.progressEnabled,
+		RemoteWriteStorageReadURL: conf.writerReadURL,
+	}
+	proceed, err := plan.Init(planner)
```

Comment:
We should pass the config object here instead of a Plan. This will allow us to avoid field in Plan that aren't needed past init (e.g. RemoteWriteStorageReadURL, ProgressEnabled)

Reply:
The reason why I dont want to pass config object to plan is that auth (being added in upcoming commits) related stuff has nothing to do in the plan.

Reply:
I hope the recent implementation is in line with what you wanted.